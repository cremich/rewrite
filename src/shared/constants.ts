export const APPLICATION_NAME = "rewrite";
export const MESSAGE_BUS_NAME_DEV = "rewrite-events-dev";
