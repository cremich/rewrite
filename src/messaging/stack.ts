import { Tags, Stack, StackProps, Aspects } from "aws-cdk-lib";
import { AwsSolutionsChecks, NagSuppressions } from "cdk-nag";
import { Construct } from "constructs";
import { MessageBus } from "./constructs/channels/bus";
import { APPLICATION_NAME } from "../shared/constants";

export interface MessagingStackProps extends StackProps {
  messageBusName: string;
}

export class MessagingStack extends Stack {
  public readonly messageBus: MessageBus;
  constructor(scope: Construct, id: string, props: MessagingStackProps) {
    super(scope, id, props);

    this.messageBus = new MessageBus(this, "MessageBus", {
      name: props.messageBusName,
    });
    this.messageBus.withMessageLogger();

    this.applyTagging();
  }

  private applyTagging() {
    Tags.of(this).add("application", APPLICATION_NAME);
    Tags.of(this).add("component", "messaging");
  }

  public addSecurityChecks() {
    Aspects.of(this).add(new AwsSolutionsChecks({ verbose: false }));

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      `/${this.stackName}/EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3/CustomResourcePolicy/Resource`,
      [{ id: "AwsSolutions-IAM5", reason: "log retention lambda custom resource" }],
    );

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      `/${this.stackName}/AWS679f53fac002430cb0da5b7982bd2287/ServiceRole/Resource`,
      [{ id: "AwsSolutions-IAM4", reason: "log retention lambda custom resource" }],
    );

    NagSuppressions.addResourceSuppressionsByPath(
      this,
      `/${this.stackName}/AWS679f53fac002430cb0da5b7982bd2287/Resource`,
      [{ id: "AwsSolutions-L1", reason: "log retention lambda custom resource" }],
    );
  }
}
