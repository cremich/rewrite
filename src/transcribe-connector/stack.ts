import { Tags, Stack, StackProps, aws_events as events } from "aws-cdk-lib";
import { Construct } from "constructs";
import { TranscribeEventProcessing } from "./constructs/event-processing";
import { MessageBus } from "../messaging/constructs/channels/bus";
import { APPLICATION_NAME } from "../shared/constants";

export interface TranscribeConnectorStackProps extends StackProps {
  messageBusName: string;
}

export class TranscribeConnectorStack extends Stack {
  constructor(scope: Construct, id: string, props: TranscribeConnectorStackProps) {
    super(scope, id, props);
    this.applyTagging();

    const messageBus = MessageBus.fromEventBus(
      this,
      "RewriteMessageBus",
      events.EventBus.fromEventBusName(this, "RewriteEventBus", props.messageBusName),
    );
    const defaultBus = MessageBus.fromEventBus(
      this,
      "DefaultMessageBus",
      events.EventBus.fromEventBusName(this, "DefaultEventBus", "default"),
    );

    new TranscribeEventProcessing(this, "TranscribeEventProcessing", {
      eventSource: defaultBus,
      eventTarget: messageBus,
    });
  }

  private applyTagging() {
    Tags.of(this).add("application", APPLICATION_NAME);
    Tags.of(this).add("component", "transcribe-connector");
  }
}
