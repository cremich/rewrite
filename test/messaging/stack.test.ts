import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { MessagingStack } from "../../src/messaging/stack";
import { MESSAGE_BUS_NAME_DEV } from "../../src/shared/constants";

test("Snapshot for MessagingStack", () => {
  const app = new App();
  const stack = new MessagingStack(app, "rewrite-messaging-dev", {
    messageBusName: MESSAGE_BUS_NAME_DEV,
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
