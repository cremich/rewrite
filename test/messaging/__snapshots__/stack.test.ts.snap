// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for MessagingStack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": {
      "DependsOn": [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "ed6cd104ff5f101d06dae8cb2b87cc6e6d69b9a22055b467ea6cae10ff023023.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "application",
            "Value": "rewrite",
          },
          {
            "Key": "component",
            "Value": "messaging",
          },
        ],
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "application",
            "Value": "rewrite",
          },
          {
            "Key": "component",
            "Value": "messaging",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3CustomResourcePolicy5D1868C0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "logs:PutResourcePolicy",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "logs:DeleteResourcePolicy",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3CustomResourcePolicy5D1868C0",
        "Roles": [
          {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3EF8F5EE7": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3CustomResourcePolicy5D1868C0",
      ],
      "Properties": {
        "Create": {
          "Fn::Join": [
            "",
            [
              "{"service":"CloudWatchLogs","action":"putResourcePolicy","parameters":{"policyName":"rewritemessagingdevEventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E356F76C2D","policyDocument":"{\\"Statement\\":[{\\"Action\\":[\\"logs:PutLogEvents\\",\\"logs:CreateLogStream\\"],\\"Effect\\":\\"Allow\\",\\"Principal\\":{\\"Service\\":\\"events.amazonaws.com\\"},\\"Resource\\":\\"",
              {
                "Fn::GetAtt": [
                  "MessageBusLogGroupDBD62FE8",
                  "Arn",
                ],
              },
              "\\"}],\\"Version\\":\\"2012-10-17\\"}"},"physicalResourceId":{"id":"EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3"}}",
            ],
          ],
        },
        "Delete": "{"service":"CloudWatchLogs","action":"deleteResourcePolicy","parameters":{"policyName":"rewritemessagingdevEventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E356F76C2D"},"ignoreErrorCodesMatching":"ResourceNotFoundException"}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
        "Update": {
          "Fn::Join": [
            "",
            [
              "{"service":"CloudWatchLogs","action":"putResourcePolicy","parameters":{"policyName":"rewritemessagingdevEventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E356F76C2D","policyDocument":"{\\"Statement\\":[{\\"Action\\":[\\"logs:PutLogEvents\\",\\"logs:CreateLogStream\\"],\\"Effect\\":\\"Allow\\",\\"Principal\\":{\\"Service\\":\\"events.amazonaws.com\\"},\\"Resource\\":\\"",
              {
                "Fn::GetAtt": [
                  "MessageBusLogGroupDBD62FE8",
                  "Arn",
                ],
              },
              "\\"}],\\"Version\\":\\"2012-10-17\\"}"},"physicalResourceId":{"id":"EventsLogGroupPolicyrewritemessagingdevMessageBuscatchrule5951C1E3"}}",
            ],
          ],
        },
      },
      "Type": "Custom::CloudwatchLogResourcePolicy",
      "UpdateReplacePolicy": "Delete",
    },
    "MessageBusCustomEventBusF105E0DD": {
      "Properties": {
        "Name": "rewrite-events-dev",
      },
      "Type": "AWS::Events::EventBus",
    },
    "MessageBusLogGroupDBD62FE8": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "RetentionInDays": 7,
        "Tags": [
          {
            "Key": "application",
            "Value": "rewrite",
          },
          {
            "Key": "component",
            "Value": "messaging",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "MessageBuscatchrule43EBB77C": {
      "Properties": {
        "EventBusName": {
          "Ref": "MessageBusCustomEventBusF105E0DD",
        },
        "EventPattern": {
          "source": [
            {
              "prefix": "",
            },
          ],
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition",
                  },
                  ":logs:",
                  {
                    "Ref": "AWS::Region",
                  },
                  ":",
                  {
                    "Ref": "AWS::AccountId",
                  },
                  ":log-group:",
                  {
                    "Ref": "MessageBusLogGroupDBD62FE8",
                  },
                ],
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
